dojo/tools/jfrog_xray_api_summary_artifact/parser.py

```diff
@@ -74,22 +69,20 @@ def get_item(vulnerability, service, test):
     if len(impact_paths) > 0:
         impact_path = decode_impact_path(impact_paths[0])
 
-    # The unique_id_from_tool is set only when a given component (SHA) has a specific unique Finding (XRAY or CVE)
     if 'issue_id' in vulnerability:
-        title = vulnerability['issue_id'] + " - " + impact_path.name + ":" + impact_path.version
-        unique_id_from_tool = vulnerability['issue_id'] + " " + impact_path.sha
+        unique_id_from_tool = str(hash(impact_path.name + impact_path.version + vulnerability['issue_id']))
```

Comment:
The unique id from the tool has been corrected to depict a unique hash that is derived from the affected component at a  given version and the XRAY id, CVE, or description (if Xray does not provide an id). Hash code might also have a performance improvement impact when discovering duplicates and decupling.

Reply:
shouldn't the unique_id_from_tool  be `vulnerability['issue_id']` ?

Reply:
I totally agree and it is the first priority, unfortunately, not all results from JFrog Xray have an XRAY issue_id. There are 3 cases that need to be considered in the respective order of preference :

1. JSON from XRay includes an XRAY-xxxxxx id (issue_id)

![image](https://user-images.githubusercontent.com/127124/204474398-a6ac778a-af31-4af9-b38d-27936c73d0d9.png)

2. No XRAY-xxxxxx id exists in the results -> use the CVE if it exists

![image](https://user-images.githubusercontent.com/127124/204474558-94590197-d409-463e-9874-0dfc7ece8785.png)

3. Neither the issue_id exists nor a CVE -> use the finding description
 
![image](https://user-images.githubusercontent.com/127124/204474925-4262b6f5-4691-48e4-87fa-c562e855deca.png)

Reply:
For the last case the unique id from tool is a hash that combines the component name, version and the title.

![image](https://user-images.githubusercontent.com/127124/204476571-97a1a3a6-ef97-4adc-97f2-2e65a0e6627a.png)

One question that I need to ask you @damiencarol is in regard to the definition of the unique_id_from_tool. For **artifacts** (i.e. microservices) that have findings in dependant **components** is it:

_( NOTE : Available ID = XRAY, CVE or title of finding 

or reference let's consider that we have the following cases, there the only thing that differs is the artifact:

Finding 1 : **Microservice A**, Available ID A, Component A, Version A
Finding 2 : **Microservice B**, Available ID A, Component A, Version A

)_

a) a unique finding that refers to a specific **component** at a specific **version**. In this case, they will have the same unique_id_from_tool 

b) a unique finding that refers to a specific **component** at a specific **version** AND in relation to a specific **artifact**. In this case, they will have a different unique_id_from_tool

Reply:
I would think option B would be the preferred choice. It will no doubt allow for more findings to be created (less duplicates identified) but I would much prefer that to fixing a vuln in Microservice A, and then forgetting about Microservice B since that was identified as a duplicate.

Reply:
I agree with you @Maffooch and will alter the creation of the unique id to include the **Artifact**. 

Regarding the resolution of **Findings**. If one has the same **Finding** across multiple **Artifacts** (Microservices) it would make sense to store this information somewhere for the developers to have an easier way to group them and address them by bumping up the **Component** dependencies or working on a specific **Artifact** at the time.

For now, the only way I see is to use the **Location**, unless there is a way that I am not aware of. Maybe the inclusion of an **Artifact** entity can be considered?

Reply:
In fact, I will opt to use the sha_256 key of the artifact to make sure that the exact binaries are identical for a given **Artifact** and not only the name and version.

![image](https://user-images.githubusercontent.com/127124/205254874-5d3da53a-f581-4b0b-b83b-ab9bdd3e4b12.png)